components/datadog/agent/docker_image.go

```diff
@@ -14,26 +13,43 @@ const (
 	defaultAgentImageRepo        = "gcr.io/datadoghq/agent"
 	defaultClusterAgentImageRepo = "gcr.io/datadoghq/cluster-agent"
 	defaultAgentImageTag         = "latest"
+	defaultOTAgentImageTag       = "7-ot-beta"
 )
 
-func dockerAgentFullImagePath(e config.Env, repositoryPath, imageTag string) string {
+func dockerAgentFullImagePath(e config.Env, repositoryPath, imageTag string, otel bool) string {
 	// return agent image path if defined
 	if e.AgentFullImagePath() != "" {
 		return e.AgentFullImagePath()
 	}
 
 	// if agent pipeline id and commit sha are defined, use the image from the pipeline pushed on agent QA registry
-	if e.PipelineID() != "" && e.CommitSHA() != "" {
-		return utils.BuildDockerImagePath(fmt.Sprintf("%s/agent", aws.AgentQAECR), fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA()))
+	if e.PipelineID() != "" && e.CommitSHA() != "" && imageTag == "" {
+		tag := fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA())
+		if otel {
+			tag = fmt.Sprintf("%s-7-ot-beta", tag)
+		}
+
+		exists, err := e.InternalRegistryImageTagExists(fmt.Sprintf("%s/agent", e.InternalRegistry()), tag)
+		if err != nil || !exists {
+			panic(fmt.Sprintf("image %s/agent:%s not found in the internal registry", e.InternalRegistry(), fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA())))
+		}
+		return utils.BuildDockerImagePath(fmt.Sprintf("%s/agent", e.InternalRegistry()), fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA()))
 	}
 
 	if repositoryPath == "" {
 		repositoryPath = defaultAgentImageRepo
 	}
+
+	if imageTag == "" && otel {
+		imageTag = defaultOTAgentImageTag
+	}
+
 	if imageTag == "" {
 		imageTag = dockerAgentImageTag(e, config.AgentSemverVersion)
 	}
 
+	fmt.Println("USING THE FOLLOWING IMAGE:" + utils.BuildDockerImagePath(repositoryPath, imageTag))
+
 	return utils.BuildDockerImagePath(repositoryPath, imageTag)
```

Comment:
Is the log needed? If yes, you could store the result of calling `utils.BuildDockerImagePath(repositoryPath, imageTag)` twice

Reply:
Was just for me to debug, I will delete it